toolbox/__init__.py

```diff
@@ -239,17 +238,19 @@ def vocoder_progress(i, seq_len, b_size, gen_rate):
 
         #Enable replay and save buttons:
         self.ui.replay_wav_button.setDisabled(False)
-        self.ui.save_wav_button.setDisabled(False)
+        self.ui.export_wav_button.setDisabled(False)
 
         #Update waves combobox
-        self.current_wav = wav
-        if len(self.waves_list) == MAX_WAVES:
+        self.waves_count += 1
+        if self.waves_count >= MAX_WAVES:
           self.waves_list.pop(0)          
         self.waves_list.append(wav)
-        n_waves=len(self.waves_list)
         #TODO better naming for the combobox items?
-        self.ui.waves_list_model.setStringList(["%d" % (i+1) for i in range(n_waves)])
-        self.ui.waves_cb.setCurrentIndex(n_waves-1)
+        cb_items = ["%d" % (self.waves_count - i) for i in range(0, min(self.waves_count, MAX_WAVES))]
```

Comment:
What I intended to suggest is to maintain a list called `self.waves_namelist` and append the count to the list (as a string) every time a wave is made.

This will make it easier to implement the TODO.

Reply:
I am already doing on line 250:

`        self.waves_namelist = ["%d" % (self.waves_count - i) for i in range(0, min(self.waves_count, MAX_WAVES))]
`

I am recreating the list every time, because this is exactly what goes on the combobox, and I wouldn't be able to just keep appending to it unless if poping 0 like in line 247 with self.waves_list and then reverting this one but I don't actually see how is that going to help on the todo. Is that what you mean anyway?

What I meant by that TODO is that maybe there is a better way to name those or let the user name them.